plugins/pulp_rpm/plugins/importers/yum/existing.py

```diff
@@ -125,18 +125,17 @@ def check_all_and_associate(wanted, conduit, config, download_deferred, catalog)
     sorted_units = _sort_by_type(wanted.iterkeys())
     for unit_type, values in sorted_units.iteritems():
         model = plugin_api.get_unit_model_by_id(unit_type)
-        # FIXME "fields" does not get used, but it should
-        # fields = model.unit_key_fields + ('_storage_path',)
+        extra_fields = ('_storage_path', 'filename', '_content_type_id', 'id', 'downloaded',)
+        fields = model.unit_key_fields + extra_fields
         unit_generator = (model(**unit_tuple._asdict()) for unit_tuple in values.copy())
-        for unit in units_controller.find_units(unit_generator):
+        for unit in units_controller.find_units(unit_generator, fields=fields):
             is_rpm_drpm_srpm = unit_type in rpm_drpm_srpm
             file_exists = unit._storage_path is not None and os.path.isfile(unit._storage_path)
             if is_rpm_drpm_srpm:
                 # no matter what is the download policy, if existing unit has a valid storage_path,
                 # we need to set the downloaded flag to True
                 if file_exists and not unit.downloaded:
-                    unit.downloaded = True
-                    unit.save()
+                    unit.update(downloaded=True)
```

Comment:
i don't think we can use update here because it is only on the pre-save hook we change the _last_updated field

Reply:
I can change it back to use "save()". The reason that I change it to "update()" was to avoid updating the fields that are not included because I was worried about whether "save()" is clever enough to avoid this or not. I did a test just now and it seems like my worry is unnecessary.

Reply:
Done. changed it back to use save()